### ISSUE TITLE
Ctrl+U opens index.html in a text editor

### ISSUE THREAD

**Issue by colltoaction** [FPGAwars/icestudio#767]
Hi! I'm seeing an `index.html` editor window every time I hit Ctrl+U. This doesn't happen with the Upload menu button or other shortcuts such as Ctrl+B.

I think the issue comes from unfinished work around here, or maybe with the shortcut settings.

https://github.com/FPGAwars/icestudio/blob/63c6c71a3c705f88ff289935f129ffdb34072ea3/app/scripts/services/utils.js#L1100-L1109

I'm on Icestudio 0.12 on Linux.

Thanks for taking a look, maybe I can fix it with some guidance.

**Comment by cavearr** [comment]
Hi Martin! could you attach an screenshot with the editor window opened after Ctrl+U?

I can't reproduce your problem and i need to view how is happening to fix it or help you to fix it.

Thanks!

**Comment by colltoaction** [comment]
Sure!

![screenshot](https://github.com/user-attachments/assets/e0d05e5d-3f2e-4e67-8e58-d8b3038745b7)

**Comment by cavearr** [comment]
WTF!!! 😮 i think you have some shortcut in your system linked to open an external editor. I'll try to research it and tell you.

what's your linux distribution? 

could you try if in Chrome browser and in firefox if you press ctrl+u in any page the editor appears?

Thanks!

**Comment by colltoaction** [comment]
😂 I'm using the AppImage in Ubuntu 23.04 with Firefox and I have not installed Chrome. The firefox shortcut for view source is Ctrl+U as well.

**Comment by cavearr** [comment]
Yes i think the Operating system takes de control over ctrl+U for open editor in each application that could be editable XD

I look for a solution to this, thanks for detect it! it's very weird, i'm telling you soon

**Comment by colltoaction** [comment]
Thanks! Tengo unas cositas más de UX si querés que te pase pero mirá que soy muy detallista jaja.

**Comment by cavearr** [comment]
Todo el feedback es bueno tu lanza ideas se evalúan y se mira a ver. Yo ando preparando un gran set de cambios a nivel de ui y motor de cara al otoño, puede ser buen momento para nuevas ideas por si pudieran encajar.

**Comment by TimRudy** [comment]
I noticed that earlier too - on Windows/Chrome - I didn’t research it, but yes, it would be interesting to fix it and understand what the cause was.

**Comment by cavearr** [comment]
It appears that this shortcut is used by the operating system to open the code editor configured on the system by default (if you have installed and configured-linked the editor on the system). I haven't found any solution yet, I think it might be impossible to replace this (I have to try some ideas tomorrow but I think it might be a limitation). 

If the limitation is confirmed, I think it might be a good idea to change this shortcut (e.g. Ctrl+Shit+U or WindowsKey+U).

**Comment by cavearr** [comment]
@lmcapacho fix at #793 and this feature is available in the last wip https://downloads.icestudio.io
